src/mapfile2js/parse/resolveSymbolset.ts

```diff
@@ -25,6 +26,43 @@ function substituteSymbols(obj: any): void {
   }
 }
 
+/**
+ * Parses the Mapfile data if it contains single SYMBOL tags and replaces the name with the filename within the
+ * symbolizer tag
+ *
+ * @param {Mapfile} mapfile Parsed Mapfile Object
+ */
+export function resolveSymbolsFromMapfile(mapfile: Mapfile): Mapfile {
```

Comment:
I think this could be merged with the above function `substituteSymbols`

Reply:
The function `substituteSymbols` is unfortunately only executed if the definition of the symbols used is in a separate file within the `<SYMBOLSET>` tag. Symbols that are defined directly in the mapfile without the `<SYMBOLSET>` tag are now also taken into account.